pythonforandroid/archs.py

```diff
@@ -112,6 +112,7 @@ def get_env(self, with_flags_in_cc=True):
         env['AR'] = '{}-ar'.format(command_prefix)
         env['RANLIB'] = '{}-ranlib'.format(command_prefix)
         env['LD'] = '{}-ld'.format(command_prefix)
+        env['LDSHARED'] = env['CC'] + " -shared "
```

Comment:
Because it seemed to me like a fundamental issue with the build environment. The problem appeared in `python setup.py build_ext`, to be more specific, in `distutils/unixccompiler.py` in function `def link` when it accesses `self.linker_so` - that one simply points to the wrong compiler/linker. Of course I could hack around that in the recipe, but since the linker is simply set wrong, I felt like this should be solved in the core

Reply:
OK, I didn't investigate it that much, but my fear is that it could conflict with eventually other recipes options e.g. `-static`. I usually don't dare to touch the core that much unless I see myself having to add every time the same option to a wide range of recipes.

Reply:
Well it's `LDSHARED`, why would that ever be used for a static link? It's literally called through `link_shared_object` in `distutils/ccompiler.py`. I can see why this change would need some more testing though - but I'm all for fixing it properly, even if that will delay inclusion a little

Reply:
Oh yes sorry I overlooked, I thought it was `env['CC'] += ' -shared'`. Then yes it may make total sense, thanks.
But I still wonder if it doesn't it's PR on its own, so we also clean all the other `-shared` added around per recipes. We have a ton of them.
In fact it's been a while that I'm thinking about unit testing it at functional level. Not actually building the recipes, but checking the return of `get_recipe_env()` across recipes and cleaning it up. I'm not sure how feasible it is.

Reply:
Maybe for doing that, an integration test might be a good idea. How about setting up a test script that will do `p4a ... --requirements=python3crystax,<packagename>` for each package independently that has a recipe? Of course that would take a really long time to run, but one could still let it run overnight to get a basic overview over which recipes are currently broken or not

Reply:
Yes I also believe an integration test is safer, even though it doesn't cover everything. But as you stated it would run for ages. We are doing some continuous integration testing for only few critical recipes and that already takes ages. Not to mention flaky tests https://github.com/kivy/python-for-android/issues/1306
If you take a look at the current CI on Travis, we're running quite some builds in parallel and it still take ages. We're still very early in p4a CI so I'm sure there's room for improvement and help is more than welcome.

Reply:
Ok so should I do a separate `LDSHARED` pull request? Or just add `env["LDSHARED"] = env["CC"] + " -shared "` to the recipe? Or keep things like they are? I'm open for everything, but in any case, a proper value of `LDSHARED` does seem to be a vital prerequisite for reportlab's setup.py to run through

Reply:
I would be tempted to say I'm fine either ways. But since I honestly don't know where it should really fit (`recipe.py` could also be a candidate right?) I would say let's address it later in a dedicated issue, I've just created here https://github.com/kivy/python-for-android/issues/1360
Until then you can pull that `get_recipe_env()` quick fix I've mentioned in the ticket.
It's just that I think it requires some dedicated attention, it's a long standing technical debt that we should definitely tackle. What do you think?

Reply:
I think `arch.py` is the perfect place because that's where env["CC"] and all other things are set up as well. And I don't see how anything else outside of recipes would require a *broken* `env["LDSHARED"]` value, so it seems like the proper place to add this.